Kill all compiler warnings

Let's kill them all and enable -Werror for GCS.

Current state with GCC/Linux x64:
```
mike@mike-desktop:~/Dev/dronin$ make all_clean
[ ! -d "/home/mike/Dev/dronin/build" ] || rm -r "/home/mike/Dev/dronin/build"
mike@mike-desktop:~/Dev/dronin$ make -j8 all 1> /dev/null
/home/mike/Dev/dronin/make/firmware-common.mk:167: warning: overriding commands for target `/home/mike/Dev/dronin/build/sim_posix/sim_posix.elf_size'
Makefile.posix:359: warning: ignoring old commands for target `/home/mike/Dev/dronin/build/sim_posix/sim_posix.elf_size'
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c: In function 'OnScreenDisplayInitialize':
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1623:41: warning: passing argument 1 of 'OnScreenDisplaySettingsConnectCallback' from incompatible pointer type
  OnScreenDisplaySettingsConnectCallback(OnScreenSettingsUpdatedCb);
                                         ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:66:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaysettings.h:93:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplaySettingsConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplaySettingsHandle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1626:45: warning: passing argument 1 of 'OnScreenDisplayPageSettingsConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettingsConnectCallback(OnScreenPageUpdatedCb);
                                             ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:67:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettingsConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettingsHandle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1627:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings2ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings2ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:68:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings2.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings2ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings2Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1628:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings3ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings3ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:69:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings3.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings3ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings3Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1629:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings4ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings4ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:70:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings4.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings4ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings4Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
Project MESSAGE: tlfw_resource.qrc not found.  Automatically firmware updates disabled.
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc:387:6: warning: unused parameter ‘sig’ [-Wunused-parameter]
 bool ExceptionHandler::HandleSignal(int sig, siginfo_t* info, void* uc) {
      ^
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc: In member function ‘bool google_breakpad::ExceptionHandler::SimulateSignalDelivery(int)’:
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc:424:24: warning: missing initializer for member ‘siginfo_t::si_signo’ [-Wmissing-field-initializers]
   siginfo_t siginfo = {};
                        ^
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc:424:24: warning: missing initializer for member ‘siginfo_t::si_errno’ [-Wmissing-field-initializers]
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc:424:24: warning: missing initializer for member ‘siginfo_t::si_code’ [-Wmissing-field-initializers]
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc:424:24: warning: missing initializer for member ‘siginfo_t::_sifields’ [-Wmissing-field-initializers]
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/crash_generation/crash_generation_client.cc: In member function ‘bool google_breakpad::CrashGenerationClient::RequestDump(const void*, size_t)’:
/home/mike/Dev/dronin/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/crash_generation/crash_generation_client.cc:78:218: warning: suggest braces around empty body in an ‘if’ statement [-Wempty-body]
   IGNORE_RET(HANDLE_EINTR(sys_read(fds[0], &b, 1)));
                                                                                                                                                                                                                          ^
In file included from .uic/ui_stabilization.h:32:0,
                 from /home/mike/Dev/dronin/ground/gcs/src/plugins/config/configstabilizationwidget.h:30,
                 from /home/mike/Dev/dronin/ground/gcs/src/plugins/config/configstabilizationwidget.cpp:27:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/expocurve.h:33:0: warning: "QWT_DLL" redefined [enabled by default]
 #define QWT_DLL
 ^
<command-line>:0:0: note: this is the location of the previous definition
In file included from /home/mike/Dev/dronin/ground/gcs/src/plugins/config/expocurve.cpp:27:0:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/expocurve.h:33:0: warning: "QWT_DLL" redefined [enabled by default]
 #define QWT_DLL
 ^
<command-line>:0:0: note: this is the location of the previous definition
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configosdwidget.cpp: In member function ‘void ConfigOsdWidget::setCustomText()’:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configosdwidget.cpp:350:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i=0; i<OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; i++){
                                              ^
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configosdwidget.cpp: In member function ‘void ConfigOsdWidget::getCustomText()’:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configosdwidget.cpp:363:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i=0; i<OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; i++){
                                              ^
In file included from .uic/ui_stabilization.h:32:0,
                 from .moc/debug-shared/../../../../../../../../ground/gcs/src/plugins/config/configstabilizationwidget.h:30,
                 from .moc/debug-shared/moc_configstabilizationwidget.cpp:9:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/expocurve.h:33:0: warning: "QWT_DLL" redefined [enabled by default]
 #define QWT_DLL
 ^
<command-line>:0:0: note: this is the location of the previous definition
In file included from .moc/debug-shared/moc_expocurve.cpp:9:0:
.moc/debug-shared/../../../../../../../../ground/gcs/src/plugins/config/expocurve.h:33:0: warning: "QWT_DLL" redefined [enabled by default]
 #define QWT_DLL
 ^
<command-line>:0:0: note: this is the location of the previous definition
In file included from .uic/ui_stabilization.h:32:0,
                 from /home/mike/Dev/dronin/ground/gcs/src/plugins/config/configstabilizationwidget.h:30,
                 from /home/mike/Dev/dronin/ground/gcs/src/plugins/config/configgadgetwidget.cpp:34:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/expocurve.h:33:0: warning: "QWT_DLL" redefined [enabled by default]
 #define QWT_DLL
 ^
<command-line>:0:0: note: this is the location of the previous definition
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configoutputwidget.cpp: In member function ‘void ConfigOutputWidget::startESCCalibration()’:
/home/mike/Dev/dronin/ground/gcs/src/plugins/config/configoutputwidget.cpp:345:33: warning: unused variable ‘butt’ [-Wunused-variable]
     QMessageBox::StandardButton butt;
                                 ^
```

--- a/ground/gcs/gcs.pri
+++ b/ground/gcs/gcs.pri
@@ -126,6 +126,9 @@ linux-g++* {
     # Bail out on non-selfcontained libraries. Just a security measure
     # to prevent checking in code that does not compile on other platforms.
     QMAKE_LFLAGS += -Wl,--allow-shlib-undefined -Wl,--no-undefined
+
+    # Enable -Werror on Linux, should do this for all platforms once warnings are all eliminated
+    QMAKE_CXXFLAGS_WARN_ON += -Werror
 }
 
 win32 {

--- a/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc
+++ b/ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc
@@ -385,6 +385,8 @@ int ExceptionHandler::ThreadEntry(void *arg) {
 // This function runs in a compromised context: see the top of the file.
 // Runs on the crashing thread.
 bool ExceptionHandler::HandleSignal(int sig, siginfo_t* info, void* uc) {
+  (void)sig;
+
   if (filter_ && !filter_(callback_context_))
     return false;
 
@@ -421,7 +423,8 @@ bool ExceptionHandler::HandleSignal(int sig, siginfo_t* info, void* uc) {
 // This is a public interface to HandleSignal that allows the client to
 // generate a crash dump. This function may run in a compromised context.
 bool ExceptionHandler::SimulateSignalDelivery(int sig) {
-  siginfo_t siginfo = {};
+  siginfo_t siginfo;
+  memset(&siginfo, 0, sizeof(siginfo_t));
   // Mimic a trusted signal to allow tracing the process (see
   // ExceptionHandler::HandleSignal().
   siginfo.si_code = SI_USER;

--- a/ground/gcs/src/libs/libcrashreporter-qt/breakpad/common/linux/ignore_ret.h
+++ b/ground/gcs/src/libs/libcrashreporter-qt/breakpad/common/linux/ignore_ret.h
@@ -35,6 +35,6 @@
 // the call fails, IGNORE_RET() can be used to mark the return code as ignored.
 // This avoids spurious compiler warnings.
 
-#define IGNORE_RET(x) do { if (x); } while (0)
+#define IGNORE_RET(x) do { if (x) {} } while (0)
 
 #endif  // COMMON_LINUX_IGNORE_RET_H_

--- a/ground/gcs/src/plugins/config/configosdwidget.cpp
+++ b/ground/gcs/src/plugins/config/configosdwidget.cpp
@@ -345,10 +345,10 @@ void ConfigOsdWidget::handle_button_3_2()
 void ConfigOsdWidget::setCustomText()
 {
     const QString text = ui->le_custom_text->displayText();
-    int n_string = text.size();
+    unsigned int n_string = text.size();
 
-    for (int i=0; i<OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; i++){
-        if (i < n_string){
+    for (unsigned int i = 0; i < OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; ++i) {
+        if (i < n_string) {
             osdSettingsObj->setCustomText(i, (quint8)(text.data()[i].toLatin1()));
         } else {
             osdSettingsObj->setCustomText(i, 0);
@@ -360,7 +360,7 @@ void ConfigOsdWidget::getCustomText()
 {
     char text[OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM];
 
-    for (int i=0; i<OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; i++){
+    for (unsigned int i = 0; i < OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM; ++i) {
         text[i] =  osdSettingsObj->getCustomText(i);
     }
     QString q_text = QString::fromLatin1(text, OnScreenDisplaySettings::CUSTOMTEXT_NUMELEM);

--- a/ground/gcs/src/plugins/config/configoutputwidget.cpp
+++ b/ground/gcs/src/plugins/config/configoutputwidget.cpp
@@ -342,7 +342,6 @@ void ConfigOutputWidget::startESCCalibration()
     mbox.setText(QString(tr("Starting ESC calibration.<br/><b>Please remove all propellers and disconnect battery from ESCs.</b>")));
     mbox.setStandardButtons(QMessageBox::Ok | QMessageBox::Cancel);
     mbox.setDefaultButton(QMessageBox::Cancel);
-    QMessageBox::StandardButton butt;
     
     if (mbox.exec() != QMessageBox::Ok) return;
 

--- a/ground/gcs/src/plugins/config/expocurve.h
+++ b/ground/gcs/src/plugins/config/expocurve.h
@@ -30,7 +30,9 @@
 
 
 #include <QWidget>
-#define QWT_DLL
+#ifndef QWT_DLL
+    #define QWT_DLL
+#endif
 #include "qwt/src/qwt.h"
 #include "qwt/src/qwt_plot.h"
 #include "qwt/src/qwt_plot_curve.h"

